### PR TITLE
Separate connector upgrade from import

### LINKED ISSUE #5684: Separate connector definition update from the config import
## Tell us about the problem you're trying to solve
- Currently we automatically update connector definition when a config archive is imported.
- This makes the code very complicated. Also we probably want to update the connector definitions more frequently.

## Describe the solution you’d like
- Move the update logic out of the config import process.
- Maybe implement automatic update in the server.

## Describe the alternative you’ve considered or used
N/A

## Additional context
N/A

## Are you willing to submit a PR?
Yes.

### PR DESCRIPTION
**Pull request by tuliren**
## What
- Currently we update connector definition when the config archive is imported in the `importConfigsFromArchive` method. This makes the code complicated and buggy (see [thread](https://airbytehq.slack.com/archives/C0229LM09T4/p1631227911033300?thread_ts=1631227755.032200&cid=C0229LM09T4)).
- This PR removes the definition update logics from the `importConfigsFromArchive` method, and instead do that with the new `loadData` method separately.
- This PR resolves #5684.
- [x] This PR depends on #5964. Otherwise, the unit test will fail.

## How
- Call the `loadData` method after the configs are imported.

## Recommended reading order
1. `ConfigDumpImporter.java`

### CODE REVIEW

File: airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImporter.java [hunk 1]
```diff
@@ -140,8 +137,9 @@ public void importDataWithSeed(String targetVersion, File archive, ConfigPersist
       // 3. Import Postgres content
       importDatabaseFromArchive(sourceRoot, targetVersion);
 
-      // 4. Import Configs
-      importConfigsFromArchive(sourceRoot, seedPersistence, false);
+      // 4. Import Configs and update connector definitions
+      importConfigsFromArchive(sourceRoot, false);
+      configRepository.loadData(seedPersistence);
```
Comment (jrhizor) [inline airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImporter.java:142]: Maybe I'm missing something, but doesn't `loadData` need to handle the upgrade checks that we were handling in `importConfigsFromArchive` before to ensure we aren't overwriting source and destination definitions?

**Review by jrhizor — CHANGES_REQUESTED** [review]

**Review by airbyte-jenny — APPROVED** [review]

File: airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImporter.java (hunk 1 above)
Comment (tuliren) [reply airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImporter.java:142]: Yes, `DatabaseConfigPersistence#loadData` does avoid updating a connector if it is used. The `FileSystemConfigPersistence#loadData` overwrites everything, but it is not used. Let me change it to throw an exception.

These existing tests ensure that we don’t update a connector definition if it is in use:
https://github.com/airbytehq/airbyte/blob/master/airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceLoadDataTest.java
https://github.com/airbytehq/airbyte/blob/master/airbyte-server/src/test/java/io/airbyte/server/handlers/ArchiveHandlerTest.java#L163

**Review by jrhizor — APPROVED** [review]
Changing to approve. Didn't realize that in a more recent PR the `loadData` behavior changed. Thanks @tuliren for pointing that out!

**Comment by tuliren** [comment]
The failed auto migration test is being worked on in #5984.

**Pull request merged**